cmd/transfer-pvc/transfer-pvc.go

```diff
@@ -45,24 +46,77 @@ type TransferPVCOptions struct {
 	configFlags *genericclioptions.ConfigFlags
 	genericclioptions.IOStreams
 
-	logger             logrus.FieldLogger
-	SourceContext      string
-	DestinationContext string
-	PVCName            string
-	PVCNamespace       string
-	Endpoint           string
+	logger               logrus.FieldLogger
+	SourceContext        string
+	DestinationContext   string
+	PVCName              mappedName
+	PVCNamespace         mappedName
+	DestStorageClassName string
```

Comment:
I wonder if we should convert these flags into an internal data structure like:
```
apiVersion: config.transfer-pvc.crane.dev/v1alpha1
kind: PVCConfig
spec:
    source: <ObjectReference>
    claimTemplate: <PVCTemplate>
    verify: <bool>
    ...
```

1. read the config file and decode the above values
2. if flags provided override the flag values specs
3. Then use the above object to actually transfer the PVC. 

IMO it will abstract out the input with transfer action nicely. Not suggesting doing all in this PR, open to how to break it up if this sounds like a good idea

cc @djzager not sure if you had opinions, just wanted to loop you in.

Reply:
I think, at a minimum, we can put these into a struct that represents what a `PVCConfigSpec` would be. Since I believe concrete API types like that are probably more appropriate in crane-lib than in the cli directly.

I would support us creating an issue to convert such a struct to a more clearly defined API type in crane-lib. My recommendation for this PR would be to create a new `PVCConfigSpec` with this stuff.